### PR TITLE
Feature/eigen3 bundling

### PR DESCRIPTION
**Pull request by karlnapf**
@vigsterkr @lisitsyn @besser82 please give feedback

### CODE REVIEW

**Comment by vigsterkr** [comment]
@karlnapf remove the eclipse stuff....

**Comment by karlnapf** [comment]
@vigsterkr Thanks, that slipped through somehow. Feel free to merge if you agree otherwise

**Comment by lisitsyn** [comment]
:+1:

File: CMakeLists.txt [hunk 1]
```diff
@@ -560,18 +560,13 @@ IF (ARPACK_FOUND)
 	SET(POSTLINKFLAGS ${POSTLINKFLAGS} ${ARPACK_LIB})
 ENDIF()
 
-OPTION(ENABLE_EIGEN "Enable Eigen" ON)
-OPTION(BUNDLE_EIGEN "Bundle Eigen" OFF)
-IF(BUNDLE_EIGEN)
+OPTION(BUNDLE_EIGEN "Bundle Eigen" ON)
```
Comment (vigsterkr) [inline CMakeLists.txt]: mmm let's not do this... :)
i.e. remove the whole option `BUNDLE_EIGEN` and when `NOT EIGEN3_FOUND` then just simply include the external eigen...

File: src/shogun/ui/GUIPreprocessor.cpp [hunk 2]
```diff
@@ -70,10 +69,8 @@ CPreprocessor* CGUIPreprocessor::create_pca(bool do_whitening, float64_t thresho
 		SG_ERROR("Could not create preproc PCA, do_whitening %i threshold %e", do_whitening, threshold)
 
 	return preproc;
-#else //HAVE_EIGEN3
 	SG_ERROR("Could not create preproc PCA - eigen3 not available at compile time\n")
```
Comment (vigsterkr) [inline src/shogun/ui/GUIPreprocessor.cpp]: this error doesn't make too much sense... :)

File: CMakeLists.txt [hunk 3]
```diff
@@ -770,7 +765,7 @@ ENDIF ()
 OPTION (LinAlgBackend "Set global linear algebra backend library for all modules")
 IF (LinAlgBackend STREQUAL "EIGEN3")
 	FIND_PACKAGE(Eigen3 ${EIGEN_VERSION_MINIMUM} REQUIRED)
```
Comment (vigsterkr) [inline CMakeLists.txt]: doesn't make much sense anymore, because previous it has been either found or bundled.

File: CMakeLists.txt [hunk 4]
```diff
@@ -792,7 +787,7 @@ ENDIF ()
 OPTION (CoreLib "Set linear algebra backend library for core module")
 IF (CoreLib STREQUAL "EIGEN3")
 	FIND_PACKAGE(Eigen3 ${EIGEN_VERSION_MINIMUM} REQUIRED)
```
Comment (vigsterkr) [inline CMakeLists.txt]: same here as above...

File: CMakeLists.txt [hunk 5]
```diff
@@ -770,7 +765,7 @@ ENDIF ()
 OPTION (LinAlgBackend "Set global linear algebra backend library for all modules")
 IF (LinAlgBackend STREQUAL "EIGEN3")
 	FIND_PACKAGE(Eigen3 ${EIGEN_VERSION_MINIMUM} REQUIRED)
-	IF (EIGEN3_FOUND AND ENABLE_EIGEN)
+	IF (EIGEN3_FOUND)
```
Comment (vigsterkr) [inline CMakeLists.txt]: i'm not 100% if this is set when you bundle, needs to be doublechecked...

File: CMakeLists.txt [hunk 6]
```diff
@@ -812,7 +807,7 @@ ENDIF ()
 OPTION (ReduxLib "Set linear algebra backend library for reduction module")
 IF (ReduxLib STREQUAL "EIGEN3")
 	FIND_PACKAGE(Eigen3 ${EIGEN_VERSION_MINIMUM} REQUIRED)
```
Comment (vigsterkr) [inline CMakeLists.txt]: same as above.

File: CMakeLists.txt [hunk 7]
```diff
@@ -832,7 +827,7 @@ ENDIF ()
 OPTION (LinSolverLib "Set linear algebra backend library for linear solver module")
 IF (LinSolverLib STREQUAL "EIGEN3")
 	FIND_PACKAGE(Eigen3 ${EIGEN_VERSION_MINIMUM} REQUIRED)
```
Comment (vigsterkr) [inline CMakeLists.txt]: same as above

File: CMakeLists.txt [hunk 8]
```diff
@@ -852,7 +847,7 @@ ENDIF ()
 OPTION (EigenSolver "Set linear algebra backend library for eigen solver module")
 IF (EigenSolver STREQUAL "EIGEN3")
 	FIND_PACKAGE(Eigen3 ${EIGEN_VERSION_MINIMUM} REQUIRED)
```
Comment (vigsterkr) [inline CMakeLists.txt]: again...

**Comment by karlnapf** [comment]
@vigsterkr thanks for taking a look. Addressed all of them

File: src/shogun/machine/gp/GaussianARDSparseKernel.h [hunk 9]
```diff
@@ -71,7 +71,6 @@ class CGaussianARDSparseKernel: public CGaussianARDKernel
 private:
 	void initialize_sparse_kernel();
 
-#if defined(HAVE_EIGEN3) && defined(HAVE_LINALG_LIB)
```
Comment (karlnapf) [inline src/shogun/machine/gp/GaussianARDSparseKernel.h]: @yorkerlin Hey Wu, please don't do such checks in the future. They are really hard to automatically re-factor. Rather use two lines with `#ifdef`

**Comment by vigsterkr** [comment]
@karlnapf for future reference: please do not PR a feature branch, it doesn't make any sense and it makes travis work twice.

**Pull request closed without merging**